bot/utilities/Queue.ts

```diff
@@ -38,37 +51,39 @@ export default class Queue {
     return this.entries.indexOf(this.findStudent(idToGet));
   }
 
-  enqueueStudent(idToAdd: string): void {
-    const studentToAdd: QueueEntry = {
-      id: Math.random() * 1000,
+  enqueueQueueEntryEntity(queueEntryEntity: QueueEntryEntity) {
+    const newEntry = QueueEntry.fromQueueEntryEntity(queueEntryEntity);
+    newEntry.setQueueId(this.properties.id);
+    this.entries.push(newEntry);
+  }
+
+  enqueueStudentById(idToAdd: string): void {
+    const studentToAdd = new QueueEntry({
+      id: null,
       userId: idToAdd,
       queueId: this.properties.id,
       question: "",
       resolved: false,
-      createdAt: new Date(Date.now()),
-      // leaving updatedAt for post-creation updates only
-    };
+    });
     this.entries.push(studentToAdd);
   }
 
   dequeueStudent(idToRemove: string): void {
     this.entries.splice(this.getQueuePosition(idToRemove), 1);
   }
 
-  queueToString(): String {
-    var queueString: String = "[";
-    this.entries.forEach(function (student) {
-      if (queueString.length != 1) {
-        queueString += ",";
-      }
+  propertiesToString(): string {
+    return (
+      `\n\n           id: ${this.properties.id}\n` +
+      `      ownerId: ${this.properties.ownerId}\n` +
+      `    channelId: ${this.properties.channelId}\n` +
+      `       status: ${this.properties.status}\n` +
+      `           at: ${this.properties.startTime}`
+    );
+  }
 
-      queueString += `{id: ${student.id},
-          userId: ${student.userId},
-          resolved: ${student.resolved}, 
-          createdAt: ${student.createdAt},
-          updatedAt: ${student.updatedAt ?? ""}}`;
-    });
-    queueString += "]";
-    return queueString;
+  entriesToString(): string {
+    const entryStrings = this.entries.map((entry) => entry.toString());
+    return "[" + entryStrings.join(",") + "]";
```

Comment:
![Screen Shot 2022-02-08 at 9 26 08 AM](https://user-images.githubusercontent.com/48501485/153041956-9add2d21-b199-4b01-98e1-1f469fae9b3a.png)

Think everything looks great. A slight improvement might be when there are no students in the queue that there is a message for that instead of just an empty array.

Reply:
See [845f85d](https://github.com/beaver-hours/BeaverHours/pull/47/commits/845f85d8d0c2797458497cff1703263663b885ea)

Reply:
@coucoulesr sweet thanks.